### PR TITLE
feat: add a single endpoint that visualizes a new table. Enables external superset integration.

### PR DESCRIPTION
**Pull request by bkyryliuk**
### CATEGORY

Choose one

- [ ] Bug Fix
- [x] Enhancement (new features, refinement)
- [ ] Refactor
- [ ] Add tests
- [ ] Build / Development Environment
- [ ] Documentation

### SUMMARY
This PR implements and ability for external systems to integrate with superset, more specifically it implements a single endpoint that gets or creates a table for the given database and takes you to the explore view.

### TEST PLAN
Unit tests
Used in dropbox production for almost 2 quarters

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [x] Introduces new feature or API
- [ ] Removes existing feature or API

### REVIEWERS
* @villebro  
* @mistercrunch

### CODE REVIEW

File: superset/views/core.py [hunk 1]
```diff
@@ -565,7 +565,53 @@ def import_dashboards(self) -> FlaskResponse:
...
+        # overloading is_sqllab_view to be able to hide the temporary tables from the table list.
+        is_sqllab_view = request.args.get("is_sqllab_view") == "true"
+        table_id = create_if_not_exists_table(
+            database_id, schema_name, table_name, is_sqllab_view=is_sqllab_view
+        )
+        return redirect(f"/superset/explore/{datasource_type}/{table_id}")
+
+    @event_logger.log_this
+    @has_access
+    @expose("/explore/<datasource_type>/<datasource_id>/", methods=["GET", "POST"])
```
Comment (john-bodley) [inline superset/views/core.py:614]: See note above regarding the URL converter.

**Review by john-bodley — COMMENTED** [review]
I think this feature may be a good topic for the biweekly Apache Superset meetup. I think there is merit  in having the community discuss whether this flow (which differs from the current model); explore then register datasource, as opposed to register datasource then explore, should be supported.

**Comment by mistercrunch** [comment]
I think we're currently working on the dataset REST API under the official public API at `/api/v1`

**Comment by bkyryliuk** [comment]
@mistercrunch - yep I've seen a lot of good work done for the API v1, a goal for this endpoint would be to enable integration on the url level e.g. without a need of implementing server side calls of registering the data sources.

Main idea is to provide a way to construct a url and easily embed the links in various external tools that help with Superset adoption. I'll update the description a bit and will add a discussion topic to the community meetup

File: superset/security/manager.py [hunk 2]
```diff
@@ -273,15 +273,15 @@ def can_access_datasource(self, datasource: "BaseDatasource") -> bool:
             "datasource_access", datasource.perm or ""
         )
 
-    def get_datasource_access_error_msg(self, datasource: "BaseDatasource") -> str:
+    def get_datasource_access_error_msg(self, datasource_name: Optional[str]) -> str:
```
[the remark this answers is not in the document]
Comment (bkyryliuk) [reply superset/security/manager.py:276]: That's a grey area in superset, e.g. we have database and schema level permissions that are leveraged in both explore view and sqllab. For consistency it feels to me that we should treat them the same e.g. having access to the table in sqllab and having access to the datasources that represents the same table.

**Review by villebro — COMMENTED** [review]
Functionality-wise I think this is a good feature to have, but I'd prefer to see this endpoint added to one of the new `api.py` files with all the new decorators etc, rather than the legacy `views/core.py` one. Maybe @dpgaspar could chime in here with recommendations?

**Comment by bkyryliuk** [comment]
> Functionality-wise I think this is a good feature to have, but I'd prefer to see this endpoint added to one of the new `api.py` files with all the new decorators etc, rather than the legacy `views/core.py` one. Maybe @dpgaspar could chime in here with recommendations?

happy to leverage new apis, @dpgaspar could you suggest what place / file would be a right place for this endpoint ?

**Pull request closed without merging**